ably/rest_client_test.go

```diff
@@ -195,3 +197,25 @@ var _ = Describe("RestClient", func() {
 		})
 	})
 })
+
+func TestRSC7(t *testing.T) {
+	app, err := ablytest.NewSandbox(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	c, err := ably.NewRestClient(app.Options())
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Run("must set version header", func(ts *testing.T) {
+		req, err := c.NewHTTPRequest(&ably.Request{})
+		if err != nil {
+			ts.Fatal(err)
+		}
+		h := req.Header.Get("X-Ably-Version")
```

Comment:
Most of the other test suites explicitly check that the header is sent, instead of just checking that it is set on the request. That could be done in two ways:

- start a webserver locally;
- use the echo server (https://github.com/ably/echoserver).

The second option is better but I don't see anything in there at the moment that will echo request headers, although this is easily added.

@ORBAT any thoughts?

Reply:
@paddybyers that is not really necessary. Because this is the only place throughout the library where the request is composed.

So, what you see is a stated fact. We are only checking that our facts are correct.

Reply:
Unit testing `NewHTTPRequest` like this is in my opinion a good idea, but we do probably _also_ want to have tests that actually do HTTP requests. https://golang.org/pkg/net/http/httptest/ would probably be good for building something like the echo server

Reply:
@ORBAT I agree with you. What I was trying to say is It will be an overkill to do that with this test. We are only interested to see if the header is set, and it is set within the function. If the header was set by the server that is a different story.

Reply:
My thoughts exactly